app/src/main/kotlin/com/twain/interprep/presentation/ui/components/generic/IPTextInputDeletable.kt

```diff
@@ -0,0 +1,76 @@
+package com.twain.interprep.presentation.ui.components.generic
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.twain.interprep.R
+import com.twain.interprep.data.ui.TextInputAttributes
+import com.twain.interprep.presentation.ui.theme.BackgroundDarkPurple
+
+@Composable
+fun IPTextInputDeletable(
+    modifier: Modifier = Modifier,
+    inputText: String,
+    isBackPressed: Boolean = false,
+    textInputAttributes: TextInputAttributes,
+    onTextUpdate: (text: String) -> Unit,
+    onDeleteClicked :() -> Unit,
+    showDeleteIcon : Boolean = true
+) {
+    Row(
+        modifier = Modifier
```

Comment:
When we pass modifier in the Composable, let's use that as modifier in parent layout.